source/mcpele/record_pair_dist_histogram.h

```diff
@@ -61,6 +67,25 @@ class RecordPairDistHistogram : public Action {
     }
 };
 
+template<size_t BOXDIM>
+class RecordPairDistHistogramQuench : public RecordPairDistHistogram<BOXDIM> {
+private:
+    std::shared_ptr<pele::GradientOptimizer> m_opt;
+public:
+    RecordPairDistHistogramQuench(pele::Array<double> boxvector, const size_t nr_bins, const size_t eqsteps, const size_t record_every, std::shared_ptr<pele::GradientOptimizer> opt)
+        : RecordPairDistHistogram<BOXDIM>(boxvector, nr_bins, eqsteps, record_every),
+          m_opt(opt)
+    {}
+    void process_add_configuration(pele::Array<double>& coords)
+    {
+        auto c = coords.copy();
+        m_opt->reset(c);
+        m_opt->run();
```

Comment:
In that case 
RecordPairDistHistogram
is selected instead of the newly derived
RecordPairDistHistogramQuench.
I agree that in hindsight it would have been maybe better to just have a flag and leave it in one cpp class. Especially because Python can't deal with the templates properly.